Vector/Vector.hpp

```diff
@@ -0,0 +1,294 @@
+enum { delta = 2 };
+
+template <class T>
+class Vector {
+public:
+  Vector() = delete;
+  Vector(const size_t size);
+  Vector(std::initializer_list<T> args);
+  Vector(const T *begin, const T *const end);
+  Vector(const Vector &rhs);
+  Vector(Vector &&rhs);
+  ~Vector();
+  Vector &operator=(const Vector &rhs);
+  Vector &operator=(Vector &&rhs);
+
+  T *begin() const;
+  T *end() const;
+  void push_back(const T &value);
+  void push_back(T &&value);
+  void push_front(const T &value);
+  void pop_back();
+  template <class... Arg> void emplace_back(Arg &&... args);
+
+  T *insert(const size_t pos, const T value);
+  T *erase(const size_t pos);
+  T *erase(const int *pos);
+  T *erase(const T *begin, const T *const end);
+  T back() const;
+  T front() const;
+
+  T &operator[](size_t pos);
+  const T &operator[](size_t pos) const;
+  void resize(const size_t count);
+  void reserve(const size_t new_cap);
+  size_t size() const;
+  size_t capacity() const;
+  bool empty() const;
+
+private:
+  size_t capacity_{10};
+  T *data_{};
+  size_t size_{};
+};
+
+template <class T>
+class Stack : private Vector<T> {
+public:
+  Stack() = default;
+  Stack(const size_t size) : Vector<T>(size){};
+  Stack(const Stack &rhs) : Vector<T>(rhs){};
+  Stack(Stack &&rhs) : Vector<T>(rhs){};
+  ~Stack() { Vector<T>::~Vector(); };
```

Comment:
уууууу, не так не надо, деструктор неявно вызывает деструктор базового класса
```suggestion
  ~Stack() { };
```

Reply:
Done.